src/components/search.tsx

```diff
@@ -65,16 +114,16 @@ export const Search: React.FC<Props> = ({ setSearch, setSearchType }) => {
       </Button.Group>
       <form
         onSubmit={(e) => {
-          setSearch(val);
+          setSearch(search);
           e.preventDefault();
         }}
       >
         <Input
           fluid
           placeholder="Search..."
-          value={val}
+          value={search}
           onChange={(e) => {
-            setVal(e.target.value);
+            setSearch(e.target.value);
```

Comment:
This triggers the effect and hits the backend with a request, thus it will happen on each keystroke. Use `val` and `setVal` instead.